src/MongoDB.Driver.Core/Core/Compression/Zstd/Zstandard64NativeMethods.cs

```diff
@@ -0,0 +1,270 @@
+            ThrowIfError(result);
+            return result;
+        }
+
+        // private static methods
+        private static Lazy<TDelegate> CreateLazyForDelegate<TDelegate>(string name)
+        {
+            return new Lazy<TDelegate>(() => __libraryLoader.Value.GetDelegate<TDelegate>(name), isThreadSafe: true);
+        }
+
+        private static void ThrowIfError(ulong code)
+        {
+            if (Zstandard64NativeMethods.ZSTD_isError(code))
+            {
+                var errorPtr = Zstandard64NativeMethods.ZSTD_getErrorName(code);
+                var errorMsg = Marshal.PtrToStringAnsi(errorPtr);
+                throw new IOException(errorMsg);
+            }
+        }
+
+        private static bool ZSTD_isError(ulong code)
+        {
+            return __ZSTD_isError.Value(code);
+        }
+
+        private static IntPtr ZSTD_getErrorName(ulong code)
+        {
+            return __ZSTD_getErrorName.Value(code);
+        }
+
+        // nested types
+        private class Delegates64
+        {
+            public delegate ulong ZSTD_CStreamInSize();
+            public delegate ulong ZSTD_CStreamOutSize();
+            public delegate IntPtr ZSTD_createCStream();
+
+            public delegate ulong ZSTD_DStreamInSize();
+            public delegate ulong ZSTD_DStreamOutSize();
+            public delegate IntPtr ZSTD_createDStream();
+
+            public delegate long ZSTD_maxCLevel();
+
+            public delegate ulong ZSTD_flushStream(IntPtr zcs, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer);
+            public delegate ulong ZSTD_endStream(IntPtr zcs, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer);
+            public delegate ulong ZSTD_freeCStream(IntPtr zcs);
+            public delegate ulong ZSTD_freeDStream(IntPtr zds);
+
+            public delegate ulong ZSTD_initDStream(IntPtr zds);
+
+            public delegate ulong ZSTD_decompressStream(IntPtr zds, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo inputBuffer);
+
+            public delegate ulong ZSTD_initCStream(IntPtr zcs, int compressionLevel);
+
+            public delegate ulong ZSTD_compressStream(IntPtr zcs, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo inputBuffer);
+
+            public delegate bool ZSTD_isError(ulong code);
+            public delegate IntPtr ZSTD_getErrorName(ulong code);
+        }
+
+        private class ZstandardLocator : RelativeLibraryLocatorBase
+        {
+            public override string GetLibraryRelativePath(SupportedPlatform currentPlatform)
+            {
+                switch (currentPlatform)
+                {
+                    case SupportedPlatform.Windows:
+                        return @"..\..\x64\native\windows\libzstd.dll";
```

Comment:
I do not understand at all the details about where this library is supposed to be loaded from.

I defer to you and @vincentkam on this detail.